### PR TITLE
Added json to func registry

### PR DESCRIPTION
**Pull request by jigold**

### CODE REVIEW

File: src/main/scala/is/hail/expr/FunctionRegistry.scala [hunk 1]
```diff
@@ -2934,6 +2935,18 @@ object FunctionRegistry {
...
+    .. code-block:: text
+          :emphasize-lines: 2
+
+          let v = Variant("1", 278653, "A", "T") in str(v)
+          result: "1:278653:A:T"
+    """, "x" -> "Value to be stringified.")(TTHr, stringHr)
+
+  registerDependentCode("json", { () =>
+    val t = TT.t
+    (v: Code[Any]) => CM.invokePrimitive1({(x: Any) => JsonMethods.compact(t.toJSON(x))})(v) },
```
Comment (danking) [inline src/main/scala/is/hail/expr/FunctionRegistry.scala]: let's remove the curly braces as they're unnecessary.

Comment (danking) [reply src/main/scala/is/hail/expr/FunctionRegistry.scala]: I think we can actually write this as just:

```scala
CM.invokePrimitive1((x: Any) => JsonMethods.compact(t.toJSON(x)))
```

does that compile?

File: src/main/scala/is/hail/utils/FunctionDocumentation.scala [hunk 2]
```diff
@@ -342,22 +342,6 @@ object FunctionDocumentation {
               d = index(a, genename) in global.gene_dict["gene1"]
 
           result: {PLI: 0.998, hits_in_exac: 1}
-      """),
-    DocumentationEntry("str", "function", None, TString,
```
Comment (danking) [inline src/main/scala/is/hail/utils/FunctionDocumentation.scala]: whoops! thanks :)

File: src/main/scala/is/hail/expr/FunctionRegistry.scala (hunk 1 above)
Comment (jigold) [reply src/main/scala/is/hail/expr/FunctionRegistry.scala]: Not without the `(v)` at the end.

Comment (danking) [reply src/main/scala/is/hail/expr/FunctionRegistry.scala]: I think I was unclear, can we replace
```
(v: Code[Any]) => CM.invokePrimitive1({(x: Any) => JsonMethods.compact(t.toJSON(x))})(v)
```

```
CM.invokePrimitive1(JsonMethods.compact(t.toJSON(_)))
```

Comment (jigold) [reply src/main/scala/is/hail/expr/FunctionRegistry.scala]: It won't compile in that form.

**Comment by danking** [comment]
Looks good.

**Pull request merged**